File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        zerosIndex = []
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    zerosIndex.append((i, j))
                    
        for index in zerosIndex:
            for row in range(len(matrix)):
                for col in range(len(matrix[0])): 
                    if row == index[0]:
                        matrix[row][col] = 0
                    
                    if col == index[1]:
                        matrix[row][col] = 0

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
      std::vector<std::pair<int, int>> zerosIndex;
        for (int i = 0; i < matrix.size(); i++) {
            for (int j = 0; j < matrix[0].size(); j++) {
                if (matrix[i][j] == 0) {
                    zerosIndex.push_back(std::make_pair(i, j));
                }
            }
        }

        for (auto index : zerosIndex) {
            for (int row = 0; row < matrix.size(); row++) {
                for (int col = 0; col < matrix[0].size(); col++) {
                    if (row == index.first || col == index.second) {
                        matrix[row][col] = 0;
                    }
                }
            }
        }  
    }
};